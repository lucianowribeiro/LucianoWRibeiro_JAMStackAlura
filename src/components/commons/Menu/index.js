import React from 'react';
import PropTypes from 'prop-types';
import Logo from '../../../theme/icons/Logo';
import Text from '../../foundation/Text';
import Button from '../Button';
import MenuWrapper from './styles/MenuWrapper';
import SwitchIcon from '../../../theme/icons/SwitchIcon';

const links = [
  {
    texto: 'Home',
    url: '/',
  },
  {
    texto: 'Perguntas Frequentes',
    url: '/faq',
  },
  {
    texto: 'Sobre',
    url: '/sobre',
  },
];
export default function Menu({
  currentPage, onCadastrarClick, onSwitchTheme, mode,
}) {
  return (
    <MenuWrapper>
      <MenuWrapper.LeftSide>
        <Logo mode={mode} />
      </MenuWrapper.LeftSide>
      <MenuWrapper.CentralSide mode={mode}>
        {links.map((link) => (
          <li key={link.url}>
            <Text color="tertiary.main" href={link.url} mode={mode}>
              {link.texto === currentPage && link.texto}
            </Text>
            <Text color="tertiary.light" href={link.url} mode={mode}>
              {link.texto !== currentPage && link.texto}
            </Text>
          </li>
        ))}
      </MenuWrapper.CentralSide>
      <MenuWrapper.RightSide>
        <Button ghost variant="secondary.main" href="/app/login" mode={mode}>
          Entrar
        </Button>
        <Button variant="primary.main" onClick={onCadastrarClick} mode={mode}>
          Cadastrar
        </Button>
        <Button ghost onClick={onSwitchTheme} mode={mode}>
          <SwitchIcon mode={mode} />
        </Button>
      </MenuWrapper.RightSide>
    </MenuWrapper>
  );
}
Menu.defaultProps = {
  currentPage: 'Home',
};

Menu.propTypes = {
  currentPage: PropTypes.string,
  onCadastrarClick: PropTypes.func.isRequired,
  onSwitchTheme: PropTypes.func.isRequired,
  mode: PropTypes.string.isRequired,
};
