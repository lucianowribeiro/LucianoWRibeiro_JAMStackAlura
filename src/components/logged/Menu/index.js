import React from 'react';
import PropTypes from 'prop-types';
import Logo from '../../../theme/icons/Logo';
import TextField from '../../forms/TextField';
import MenuWrapper, { AddIcon } from './styles/MenuWrapper';
import HomeIcon from './icons/HomeIcon';
import LikeIcon from '../../../theme/icons/LikeIcon';
import SearchIcon from './icons/SearchIcon';
import Button from '../../commons/Button';
import { LoggedPageContext } from '../../wrappers/LoggedPage/context';
import UserAvatar from '../../../theme/UserAvatar';
import SwitchIcon from '../../../theme/icons/SwitchIcon';

export default function Menu({ mode, switchTheme }) {
  const webLogged = React.useContext(LoggedPageContext);
  return (
    <MenuWrapper mode={mode}>
      <Logo mode={mode} />
      <MenuWrapper.LeftSide>
        <MenuWrapper.SearchSide>
          <TextField
            type="search"
            placeholder="Pesquisar"
            name="pesquisar"
            color="background.light"
            mode={mode}
          />
          <SearchIcon mode={mode} />
        </MenuWrapper.SearchSide>
        <Button className="openModal" variant="background.light" onClick={webLogged.tooglePost} mode={mode}>
          <AddIcon mode={mode} />
        </Button>
        <Button variant="background.light" onClick={() => webLogged.toogleFeed(false)} mode={mode}>
          <HomeIcon mode={mode} />
        </Button>
        <LikeIcon variant="background.light" isButton color="tertiary.main" mode={mode} />
        <Button className="openFeed" variant="background.light" onClick={() => webLogged.toogleFeed(true)} mode={mode}>
          <UserAvatar size="smallest" />
        </Button>
        <Button variant="background.light" onClick={switchTheme} mode={mode}>
          <SwitchIcon mode={mode} />
        </Button>
      </MenuWrapper.LeftSide>
    </MenuWrapper>
  );
}

Menu.propTypes = {
  mode: PropTypes.string.isRequired,
  switchTheme: PropTypes.func.isRequired,
};
