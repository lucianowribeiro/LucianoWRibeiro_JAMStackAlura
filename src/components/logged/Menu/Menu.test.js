import React from 'react';
/* import user from '@testing-library/user-event'; */
import { render } from '../../../infra/test/testUtils';

import Menu from './index';

describe('<HomeIcon />', () => {
  test('renders component', () => {
    const component = render(
      <Menu mode="light" switchTheme={()=> 'teste'} />,
    );
    expect(component).toMatchSnapshot();
  });
});
