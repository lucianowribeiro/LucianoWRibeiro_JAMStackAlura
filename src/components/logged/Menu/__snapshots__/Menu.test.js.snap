// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> renders component with dark theme switch theme 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  border: 0;
  cursor: pointer;
  padding: 12px 26px;
  font-weight: bold;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 400;
  line-height: 1;
  background-color: #000;
  -webkit-transition: opacity 200ms ease-in-out;
  transition: opacity 200ms ease-in-out;
  border-radius: 12px;
}

.c0:hover,
.c0:focus {
  -webkit-filter: blur(0.9px);
  filter: blur(0.9px);
}

.c0:disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.c2 {
  color: #FFFFFF;
  width: 32px;
  height: 32px;
}

.c2:hover,
.c2:visited {
  color: #F27895;
}

@media only screen and (min-width:0px) {

}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:0px) {

}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:0px) {

}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:0px) {
  .c0 {
    font-size: 12px;
    font-weight: 400;
    line-height: 1;
  }
}

@media only screen and (min-width:768px) {
  .c0 {
    font-size: 16px;
    font-weight: 400;
    line-height: 1.25;
  }
}

<button
  class="c0 swichTheme"
  mode="dark"
>
  <svg
    aria-hidden="true"
    class="c1 c2"
    fill="currentColor"
    focusable="false"
    mode="dark"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 3H7a7 7 0 100 14h6a7 7 0 100-14zm0 12a5 5 0 11.001-10.001A5 5 0 0113 15z"
    />
  </svg>
</button>
`;

exports[`<Menu /> renders component with light theme and switch theme 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  border: 0;
  cursor: pointer;
  padding: 12px 26px;
  font-weight: bold;
  border-radius: 8px;
  font-size: 12px;
  font-weight: 400;
  line-height: 1;
  background-color: #FFFFFF;
  -webkit-transition: opacity 200ms ease-in-out;
  transition: opacity 200ms ease-in-out;
  border-radius: 12px;
}

.c0:hover,
.c0:focus {
  -webkit-filter: blur(0.9px);
  filter: blur(0.9px);
}

.c0:disabled {
  cursor: not-allowed;
  opacity: 0.2;
}

.c2 {
  color: #070C0E;
  width: 32px;
  height: 32px;
}

.c2:hover,
.c2:visited {
  color: #D7385E;
}

@media only screen and (min-width:0px) {

}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:0px) {

}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:0px) {

}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:0px) {
  .c0 {
    font-size: 12px;
    font-weight: 400;
    line-height: 1;
  }
}

@media only screen and (min-width:768px) {
  .c0 {
    font-size: 16px;
    font-weight: 400;
    line-height: 1.25;
  }
}

<button
  class="c0 swichTheme"
  mode="light"
>
  <svg
    aria-hidden="true"
    class="c1 c2"
    fill="currentColor"
    focusable="false"
    mode="light"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 3H7a7 7 0 100 14h6a7 7 0 100-14zm0 12a5 5 0 11.001-10.001A5 5 0 0113 15z"
    />
  </svg>
</button>
`;
